### DISCUSSION TITLE
Is there a way to use upload multiple for a 1-n relationship? Please help me

### DISCUSSION THREAD

**Discussion by tatvn** [Laravel-Backpack/community-forum#791] [Q&A (Help)] [answered]
Hi, i want to upload multiple file for a 1-n relationship. 
In Article Model
```php
public function filePath()
    {
        return $this->hasMany(ArticleFile::class, 'article_id');
    }
```

In ArticleFile Model
```php
public function setFilePathAttribute($value)
    {
        $attribute_name = 'file_path';
        $disk = 'upload';
        $destination_path = '/storage/app/uploads/articles/' . Carbon::now()->format('Ym');
        $this->uploadMultipleFilesToDisk($value, $attribute_name, $disk, $destination_path);
    }
```
In ArticleController
```php
CRUD::addField([
            'name' => 'file_path',
            'type'  => 'upload_multiple',
            'label' => '本文ファイル',
            'disk' => 'upload',
            'upload' => true,
])
```

**Comment by karandatwani92** [comment, marked as answer]
Hey @tatvn 

You can't use the `upload_multiple` field for a 1-n relationship. It stores the values as a JSON array in the database column.

**If the relationship is important:**

You can use https://backpackforlaravel.com/docs/6.x/crud-uploaders#uploaders-for-spatie-medialibrary

👆This associates all sorts of files with **Eloquent models**

**OR**

You can use relationship subfields(title, upload), try these steps and let me know if they help!

- Field setup: https://backpackforlaravel.com/docs/6.x/crud-fields#relationship-pro
- Save additional data to the pivot table: https://backpackforlaravel.com/docs/6.x/crud-fields#save-additional-data-to-pivot-table
- Handling uploads in relationship fields: https://backpackforlaravel.com/docs/6.x/crud-uploaders#handling-uploads-in-relationship-fields


**OR**

You can create a custom field using existing `upload_multiple`.  You have to modify it & create your saving logic:
`php artisan backpack:field new_field_name --from= upload_multiple`

**OR**

Create a few accessor & mutator combos which turn JSON and save the values to a related table.
